src/probes/PrometheusClient.ts

```diff
@@ -0,0 +1,57 @@
+import * as joi from '@hapi/joi';
+import 'joi-extract-type';
+
+import { PrometheusDriver } from 'prometheus-query';
+import { Logger } from 'werelogs';
+
+const log = new Logger('breakbeat:prometheusClient');
+
+export const prometheusClientConfigSchema = joi.object({
+    endpoint: joi.string().uri({
+        scheme: ['http', 'https'],
+        allowRelative: false,
+    }).required(),
+    timeout: joi.number().optional(),
+});
+
+export type PrometheusClientConfig = joi.extractType<typeof prometheusClientConfigSchema>;
+
+type InstantResult = {
+    value: {
+        value: number,
+    },
+};
+
+export class PrometheusClient {
+    prom: PrometheusDriver;
+    q: string;
+
+    constructor(config: PrometheusClientConfig, q: string, interval?: string) {
+        this.prom = new PrometheusDriver({
+            endpoint: config.endpoint,
+            timeout:  config.timeout,
+        });
+
+        this.q = interval ?
```

Comment:
wondering about this 'interval' :
* if we want any 'ranged' metric, it can already be passed through `q`, without any special case here. Adding this special handling just opens the door to create many more (e.g. when we need max_over_time(), or rate(), ...)
* we may need this smoothing on top of 'any' metrics : e.g. use the 5min average of the instance rate... --> AFAIK prometheus cannot handle 2 ranges, and we would need to handle this smoothing/aggregation over time directly in the probe (and then the interval makes sense, IMHO)

Reply:
This `interval` parameter just makes it easier to write the aggregation we're going to write most often, I think it doesn't prevent from writing other expressions if the average is not wanted.

I opened BREAK-4 to investigate how we can make it easy to write several kinds of aggregations.